### PR TITLE
Adding specifications for sub-annual time resolution

### LINKED ISSUE #5: Define structure for subannual timeslices
This should provide a first set of commonly used subannual timeslices.

### PR DESCRIPTION
**Pull request by danielhuppmann**
This PR adds a subfolder for the sub-annual dimension of the data format, using the disaggregation level of "months" for defining a structure of README and yaml file.

closes #5

### CODE REVIEW

File: subannual/months.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# List of months
+
+January:
+   duration: 31. / 365
```
Comment (tburandt) [inline subannual/months.yaml]: Should make much of a difference for the actual content, but the additional '.' bothers me here

**Comment by tburandt** [comment]
Also, we currently face one problem of using variable representative time-series/hours. 

For example, in the current iteration of GENeSYS-MOD, the user can flexibly choose the temporal resolution when starting a model run. Hereby, a full hourly time-series is reduced by taking each nth hour (for example each 49th). This reduced series is then smoothed and rescaled to keep the characteristics of the underlying full hourly time-series. As a result of the reduction algorithm, a new time-series consisting of consecutive hours, each representing a fraction of the year, is created. 

E.g., choosing each 73th hour for the algorithm results in 120 hours/5 days; choosing each 49th hour or 169th hour are creating resulting time-series with 178 hours or 51 hours, respectively. Therefore, a simple assignment of 'summer-day' or 'summer-hour 1' is not entirely possible. A similar problem can occur when using clustered time-series compared to reduced time-series.

File: subannual/months.yaml (hunk 1 above)
Comment (danielhuppmann) [reply subannual/months.yaml]: thanks, fixed!

**Comment by arght** [comment]
We do something similar in openTEPES. The model is able to be run (by selecting an option) every 2, 3 or 4 hours but we are still using the same hourly definition like this
2030-01-01T01:00:00+01:00
2030-01-01T02:00:00+01:00
2030-01-01T03:00:00+01:00
2030-01-01T04:00:00+01:00
2030-01-01T05:00:00+01:00
2030-01-01T06:00:00+01:00
and we puick up only the hours corresponding to the "active" ones (every 2, 3 or 4)
Andrés

**Comment by danielhuppmann** [comment]
Thanks @tburandt and @arght for these comments. The simplest approach here would be to have one category for each level of temporal resolution that will actually be used in the project (not necessarily every level that *could* be implemented by each model).

I'd suggest that I merge this PR, as this defines a structure to be followed, and then either of you starts a new PR for the first temporal resolution category that you intend to use in the project - and then we discuss naming conventions etc. in that PR.

**Comment by danielhuppmann** [comment]
Merging as no objections to my earlier comment

**Pull request merged**